contracts/0.8.25/vaults/OperatorGrid.sol

```diff
@@ -0,0 +1,411 @@
+                operator: _nodeOperator,
+                shareLimit: uint96(_tiers[i].shareLimit),
+                reserveRatioBP: uint16(_tiers[i].reserveRatioBP),
+                rebalanceThresholdBP: uint16(_tiers[i].rebalanceThresholdBP),
+                treasuryFeeBP: uint16(_tiers[i].treasuryFeeBP),
+                mintedShares: 0
+            });
+            $.tiers.push(tier);
+            group_.tiersId.push(tierIndex);
+
+            emit TierAdded(
+                _nodeOperator,
+                tierIndex,
+                uint96(_tiers[i].shareLimit),
+                uint16(_tiers[i].reserveRatioBP),
+                uint16(_tiers[i].rebalanceThresholdBP),
+                uint16(_tiers[i].treasuryFeeBP)
+            );
+
+            tierIndex++;
+        }
+    }
+
+    /// @notice Registers a new vault
+    /// @param vault address of the vault
+    function registerVault(address vault) external {
+        if (vault == address(0)) revert ZeroArgument("_vault");
+
+        ERC7201Storage storage $ = _getStorage();
+        VaultTier storage vaultTier = $.vaultTier[vault];
+        if (vaultTier.tierIndex > 0) revert VaultExists();
+
+        Group storage defaultGroup = $.groups[DEFAULT_GROUP_ADDRESS];
+        if (defaultGroup.tiersId.length == 0) revert TiersNotAvailable();
+
+        uint256 tierId = defaultGroup.tiersId[0];
+        vaultTier.tierIndex = uint128(tierId);
+
+        emit VaultAdded(DEFAULT_GROUP_ADDRESS, tierId, vault);
+    }
+
+    /// @notice Request to change tier
+    /// @param _vault address of the vault
+    /// @param _tierId id of the tier
+    function requestTierChange(address _vault, uint256 _tierId) external {
+        if (_vault == address(0)) revert ZeroArgument("_vault");
+        if (msg.sender != OwnableUpgradeable(_vault).owner()) revert NotAuthorized("requestTierChange", msg.sender);
+
+        ERC7201Storage storage $ = _getStorage();
+        if ($.tiers[_tierId].operator == address(0)) revert TiersNotAvailable();
+
+        VaultTier storage vaultTier = $.vaultTier[_vault];
+        uint256 requestedTierId = vaultTier.tierIndexRequested;
+        if (requestedTierId != 0) revert RequestAlreadyExists(requestedTierId);
+        if (vaultTier.tierIndex == _tierId) revert TierAlreadySet();
+
+        vaultTier.tierIndexRequested = uint128(_tierId);
+
+        emit TierChangeRequested(_vault, _tierId);
+    }
+
+    /// @notice Approve tier change request
+    /// @param _vault address of the vault
+    function confirmTierChange(address _vault) external {
+        if (_vault == address(0)) revert ZeroArgument("_vault");
+
+        address nodeOperator = IStakingVault(_vault).nodeOperator();
+        if (msg.sender != nodeOperator) revert NotAuthorized("confirmTierChange", msg.sender);
+
+        ERC7201Storage storage $ = _getStorage();
+        VaultTier storage vaultTier = $.vaultTier[_vault];
+        uint256 requestedTierId = vaultTier.tierIndexRequested;
```

Comment:
let's check that it hasn't been front-runned (revert if mismatches with the calldata argument)

Reply:
agree, fixed